src/components/PaginationContainer.ts

```diff
@@ -0,0 +1,177 @@
+import { Component, ReactElement, createElement } from "react";
+import { findDOMNode } from "react-dom";
+import * as dijitRegistry from "dijit/registry";
+import * as classNames from "classnames";
+import * as dojoConnect from "dojo/_base/connect";
+import * as dojoAspect from "dojo/aspect";
+
+import { ListView, WrapperProps, findTargetNode, parseStyle } from "../utils/ContainerUtils";
+import { Pagination, PaginationProps } from "./Pagination";
+import { ValidateConfigs } from "../utils/ValidateConfigs";
+import { Alert } from "./Alert";
+import "../ui/Pagination.scss";
+
+interface PaginationContainerState {
+    findingListViewWidget: boolean;
+    listViewSize: number;
+    message: string;
+    offset: number;
+    hideUnusedPaging: boolean;
+    targetListView?: ListView | null;
+    targetNode?: HTMLElement | null;
+    validationPassed?: boolean;
+}
+
+interface ValidateProps {
+    listViewSize: number;
+    offset: number;
+    hideUnusedPaging: boolean;
+    targetListView?: ListView | null;
+    targetNode?: HTMLElement | null;
+}
+
+export default class PaginationContainer extends Component<WrapperProps, PaginationContainerState> {
+    private navigationHandler: object;
+    private listListViewHeight: number;
+
+    constructor(props: WrapperProps) {
+        super(props);
+
+        this.state = {
+            findingListViewWidget: true,
+            hideUnusedPaging: false,
+            listViewSize: 0,
+            message: "",
+            offset: 1
+        };
+
+        this.updateListView = this.updateListView.bind(this);
+        this.findListView = this.findListView.bind(this);
+        this.navigationHandler = dojoConnect.connect(props.mxform, "onNavigation", this , this.findListView);
+    }
+
+    public static translateMessageStatus(fromValue: number, toValue: number, maxPageSize: number): string {
+        return window.mx.ui.translate("mxui.lib.MxDataSource", "status", [ fromValue, toValue, maxPageSize ]);
+    }
+
+    render() {
```

Comment:
missing return type

Reply:
Render is a reaction function so return is not required. If its a new feature in React16 please let me know @Andries-Smit